Input radius and skin thickness

MichaelHeines opened the pull request:
I've implemented basic changes such that you can give input values for the size/shape of the nucleus. The radius keyword takes the solid sphere equivalent radius (as was happening internally before) and the skin thickness changes 't' in the Fermi2 model. 

These changes are mainly useful for the nuclear/particle physics side of the muonic atom community. With the radius, one can evaluate how precise one needs to determine the transition energy for a certain experimental radius precision. With the skin thickness, it's also possible to evaluate what systematics are induced by changes in the nuclear shape.

subindev-d reviewed (changes requested):
hi looks good to me, there are few formatting needed .
You may format the code using [`astyle`](https://astyle.sourceforge.net/) which can be installed using `sudo apt install astyle` in Ubuntu

for this project, the code can be formatted by running `astyle src/*.cpp src/*.hpp lib/*.cpp lib/*.hpp --style=google -s2 -S -n` command in the source code directory.

Once that's done I can implement the limits for the thickness and radius

MichaelHeines commented on lib/atom.cpp:
```diff
@@ -112,24 +112,35 @@ Atom::Atom(int Z, double m, int A, NuclearRadiusModel radius_model, double fc,
   if (A == -1) {
     R = -1;
-  } else {
+  } else if (radius < 0) {
```
This is something Frederik put in and I slightly tweaked. If I'm not mistaken, this check tells you whether or not an input radius is given. If it isn't given, radius = -1, which defaults to the standard methods from mudirac. I suppose one could also rewrite this as else if (radius == -1)

subindev-d replied:
Then it would be good to write it as radius == -1, less confusing. 
```suggestion
  } else if (radius == -1) {
```

subindev-d commented on lib/atom.cpp:
```diff
@@ -112,24 +112,36 @@ Atom::Atom(int Z, double m, int A, NuclearRadiusModel radius_model, double fc,
   if (radius_model == FERMI2) {
     V_coulomb = new CoulombFermi2Potential(Z, R, A);
-  } else {
+  } else if (radius_model == SPHERE) {
```
Currrenlty if the radiusl_model == 'POINT'  case is ignored, It would be cause error if the keyword `nuclear_model: POINT`.
```suggestion
  } else {
```

subindev-d commented on lib/atom.cpp:
```diff
@@ -83,7 +83,7 @@ double TransitionMatrix::totalRate() {
  * @retval
  */
-Atom::Atom(int Z, double m, int A, NuclearRadiusModel radius_model, double fc,
+Atom::Atom(int Z, double m, int A, double radius, NuclearRadiusModel radius_model, double fc,
```
documentation comment `@param  radius: ` missing for the new parameter
[reference](https://github.com/muon-spectroscopy-computational-project/mudirac/blob/1c3189358ca8928d251238adf9eb0678edb93975/lib/atom.cpp#L78)

subindev-d commented on lib/atom.cpp:
```diff
@@ -325,9 +337,9 @@ double Atom::sphereNuclearModel(int Z, int A) {
  * @retval
  */
-DiracAtom::DiracAtom(int Z, double m, int A, NuclearRadiusModel radius_model,
+DiracAtom::DiracAtom(int Z, double m, int A, double R, NuclearRadiusModel radius_model,
```
missing comments for the documentation `@param  R:`

MichaelHeines commented:
I've added the changes you requested. Besides that, I changed around some R to radius for clarity of use throughout the code. Let me know if there's anything else

subindev-d commented:
> I've added the changes you requested. Besides that, I changed around some R to radius for clarity of use throughout the code. Let me know if there's anything else

Hi currently some tests are failing, you may run the tests by `make test` command

MichaelHeines commented:
> **subindev-d ** requested changes

I'm trying to figure out why things are different. In principle they shouldn't be. I don't see any cases that would be treated differently with these changes here. Would you mind having a look?

subindev-d commented:
> > **subindev-d ** requested changes
> 
> I'm trying to figure out why things are different. In principle they shouldn't be. I don't see any cases that would be treated differently with these changes here. Would you mind having a look?

It was becasue earlier there was only `radius_model` as an optional parameter, but now we added  the `radius` parameter as well, so while the test was running it was giving value of `radius_model` to `radius` . 
I fixed the tests in in these commits: [fixing line test](https://github.com/muon-spectroscopy-computational-project/mudirac/pull/31/commits/6263d406c817800ebe0fb9df07fc1b7be77f37b3), [other testst](https://github.com/muon-spectroscopy-computational-project/mudirac/pull/31/commits/a9e6515d6db4ec5874916baf9bb2fe2c17bbc4ce).
Eventhough it works, I am looking into if there are any good practices to implement multiple optional parameters.
